### DISCUSSION TITLE
What is the purpose of the "main" flag?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [pydjinni/pydjinni#5] [General]
Hi,

What is the purpose of the "main" flag?

Example:
https://github.com/pydjinni/pydjinni/blob/main/src/pydjinni/generator/it/function/function.djinni#L3

We need this flag?

Thanks.

**Comment by jothepro** [comment]
The `main` keyword marks an interface as an entry-point. In Java, `main` interfaces automatically load the native library on the first interaction with the Java class. You can find more information about it [here](https://pydjinni.github.io/pydjinni/v1.0a2/idl/#main-interface).

**Comment by paulocoutinhox** [comment]
I was thinking it was a limitation of the parser, I hadn't noticed this feature. Very good. Thanks.
